SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -558,12 +563,32 @@ ItemPrinterRNG::ItemPrinterRNG()
         "<b>Fix Time When Done:</b><br>Fix the time after the program finishes.",
         LockMode::UNLOCK_WHILE_RUNNING, true
     )
+    , AUTO_MATERIAL_FARMING(
+        "<b>Auto material gathering:</b><br>"
+        "After using the item printer, automatically fly to North Province (Area 3) to farm materials, "
+        "then fly back to keep using the item printer.",
+        LockMode::LOCK_WHILE_RUNNING,
+        true
+    )
+    , NUM_MATERIAL_FARM_ROUNDS(
+        "<b>Number Material farm rounds:</b><br>"
+        "One round is: Doing one item printer session (as per 'Total Rounds' entered above), "
+        "then farming materials at North Provice area 3 (as per 'Number of sandwich rounds' entered below), "
+        "then flying back to the item printer.",
+        LockMode::UNLOCK_WHILE_RUNNING,
+        3
+    )

```

Comment:
This whole thing can probably use a better explanation. If I'm understanding it correctly, it's fundamentally two loops.

The inner loop will repeat the table `TOTAL_ROUNDS` number of times and run the material farmer. The outer loop then wraps all this and runs `NUM_MATERIAL_FARM_ROUNDS`.

What I'm thinking here is to actually throw all the material farming stuff under a `GroupOption` and use its checkbox to enable/disable material farming. Depending on whether it's enabled, we change the description of `TOTAL_ROUNDS` between the old text and the new one you're proposing. Likewise, you won't need to manually show/hide the material farmer options since they'll be greyed out automatically by the GroupOption.

Inside the `GroupOption` for material farming, you can add a `StaticTextOption` at the top that explains the new logic with the material farming being an outer-loop over the whole thing.

Reply:
I'm envisioning something like this: Outer loop(item printer loop -> material farming loop).

But you're right. It's currently confusing. I'll think about how to reorganize the UI.

Reply:
I moved the option that enables Material Farming to the top, to hopefully make it more clear that this an outer loop that wraps the Item printer and material farmer. I also added some `StaticTextOption` that changes text depending on whether material farming is enabled or not.